### ISSUE TITLE
play: mpv not parsing directory path correctly

### ISSUE THREAD

**Issue by enzelb** [beetbox/beets#3253]
The expected behavior of the `play` plugin should be to run `mpv` (and any `$args`) then to "paste" the path of the query.

Note that I'm also using the `raw: yes` option in my `config.yaml`. Otherwise the mpv CLI wouldn't allow any inputs (pause, scrub forward, etc.) when playing the temporary .m3u files.

I believe that the problem has somewhat to do with mpv as well, as it will only take a string in the format: `"G:\Music\Chance the Rapper\random.mp3"`, whereas from my understanding, beets will only output the byte(?)/string as `'G:\Music\Chance the Rapper\random.mp3'`. This leads to `mpv` as parsing the input as the commands:

* `mpv 'G:\Music\Chance'`
* `mpv 'the'`
* `mpv 'Rapper\random.mp3'`

Admittedly, I don't know much about python (yet), but this is my understanding of the issue. If there was some way to configure the play command option like:

```yaml
play:
    command: mpv $args $query
```
or
```yaml
play:
    command: mpv $args "$query"
```
I also attempted to do some string magic in the `play.py` to replace the `'` with `"`, but gave up due to a lack of knowledge in string/byte manipulation.

It appears that this issue is also related to #2208 and #2717 . Maybe a better `command` configuration is all that we need?

Running `mpv` though the command line

` mpv "G:\Music\Chance the Rapper\random.mp3" `

functions as intended.

____

Some relevant pages that I was going though:

* https://beets.readthedocs.io/en/v1.4.7/plugins/play.html
* https://mpv.io/manual/master/#escaping-spaces-and-other-special-characters
___
### Problem

Running this command in verbose (`-vv`) mode:

```sh
$ beet -vv play "long time II"
user configuration: C:\Users\enzel\AppData\Roaming\beets\config.yaml
data directory: C:\Users\enzel\AppData\Roaming\beets
plugin paths:
Sending event: pluginload
library database: G:\Music\musiclibrary.db
library directory: G:\Music
Sending event: library_opened
Playing 1 track.
play: executing command: mpv [b'G:\\Music\\Chance the Rapper\\10 Day\\12 Long Time II.mp3']
```
*note the `[b'` and `']` surrounding the path
___

Led to this problem:

```
Playing: G:\Music\Chance
[file] Cannot open file 'G:\Music\Chance': No such file or directory
Failed to open G:\Music\Chance.

Playing: the
[file] Cannot open file 'the': No such file or directory
Failed to open the.

Playing: Rapper\10
[file] Cannot open file 'Rapper\10': Invalid argument
Failed to open Rapper\10.

Playing: Day\12
[file] Cannot open file 'Day\12': Invalid argument
Failed to open Day\12.

Playing: Long
[file] Cannot open file 'Long': No such file or directory
Failed to open Long.

Playing: Time
[file] Cannot open file 'Time': No such file or directory
Failed to open Time.

Auto-loading profile 'extension.mp3'
Playing: II.mp3
[file] Cannot open file 'II.mp3': No such file or directory
Failed to open II.mp3.


Exiting... (Errors when loading file)
```

### Setup

* OS: Windows 10 Home 1809 (17763.475)
* Python version: 3.7.2
* beets version:  1.4.7
* Turning off plugins made problem go away (yes/no): no
* using the Cmder consule emulator

My configuration (output of `beet config`) is:

```yaml
directory: G:\Music
library: G:\Music\musiclibrary.db

import:
    copy: no
    move: yes

plugins: play smartplaylist embedart
play:
    command: mpv $args
    raw: yes
    use_folders: no
    relative_to:
    warning_threshold: 100
embedart:
    auto: yes
    ifempty: yes
    maxwidth: 0
    compare_threshold: 0
    remove_art_file: no
smartplaylist:
    relative_to:
    playlist_dir: .
    auto: yes
    playlists: []
```

**Comment by sampsyo** [comment]
Hello! This output:

```
play: executing command: mpv [b'G:\\Music\\Chance the Rapper\\10 Day\\12 Long Time II.mp3']
```

actually looks right to me. The log message there is saying that the `mpv` command is getting executed with a _single_ argument made up of that string.

Can you clarify whether you get the mpv output above _from within beets_ or whether that's what happens when you copy & paste the command line? If it's the former, this may be a Windows-specific bug. If it's the latter, then this spacing issue is probably not the cause—your shell will parse this differently from how we're invoking the tool.

**Comment by enzelbeltran** [comment]
The mpv output from above:

```Playing: G:\Music\Chance
[file] Cannot open file 'G:\Music\Chance': No such file or directory
Failed to open G:\Music\Chance.

Playing: the
[file] Cannot open file 'the': No such file or directory
Failed to open the.

etc.
```
comes from after inputting the beets play command:

```beet play "long time II"```

it just happens automatically, so we know that mpv is definitely running, but not parsing the path correctly.

![image](https://user-images.githubusercontent.com/8944717/57241080-84c8bd00-7030-11e9-96f0-2d3bd0211d9a.png)

___

Running mpv standalone:

mpv outputs the same errors with _only single quotes_:

```mpv 'G:\\Music\\Chance the Rapper\\10 Day\\12 Long Time II.mp3'```

but works just fine with double:

```mpv "G:\\Music\\Chance the Rapper\\10 Day\\12 Long Time II.mp3"```

____
I'm thinking that if beets --verbose outputs double quotes

```play: executing command: mpv [b"G:\\Music\\Chance the Rapper\\10 Day\\12 Long Time II.mp3"]```

it would work (hence why I was poking around in play.py)

**Comment by jackwilsdon** [comment]
The `[b'...']` is the Python representation of the arguments, the single/double quotes don't really change anything. beets is passing the whole path as a single argument, which means that `mpv` _should_ accept it as-is (ignoring spaces).

**Comment by enzelbeltran** [comment]
So it's looking more like an mpv limitation rather than a beets limitation. Looking though the mpv documentation, it's pretty clear that you either need to quote the path or already be in the necessary directory to call the file.

I'm thinking that it _should be possible_ to take the argument path "output", turn it into a string, join quotes on both ends, then pass this new argument after the `mpv` command.

**Comment by sampsyo** [comment]
One way to try this out would be to experimentally use some other command other than `mpv`. For example, you could try using something silly like `dir` to see if it treats the filename arguments correctly.

**Comment by enzelbeltran** [comment]
Interesting results:

1. using `dir` on C:\, same type of error

![image](https://user-images.githubusercontent.com/8944717/57248918-e8a8b100-7043-11e9-989a-19122464c183.png)

2. using `dir` on G:\, same type of error

![image](https://user-images.githubusercontent.com/8944717/57249253-bf3c5500-7044-11e9-9ae4-343088ba3b5a.png)

3. using `start mpv`, [Errno 22]: Invalid Argument

![image](https://user-images.githubusercontent.com/8944717/57248996-1beb4000-7044-11e9-854d-99c0b128ec15.png)


4. using `start`, [Errno 22]: Invalid Argument

![image](https://user-images.githubusercontent.com/8944717/57249061-3de4c280-7044-11e9-8815-97dc84514086.png)

___

Also, I tried the dir command by itself and it output:

```Directory of G:\Music\Chance the Rapper\10 Day```

not sure if that's supposed to have quotes, but it doesn't.

![image](https://user-images.githubusercontent.com/8944717/57249481-5c978900-7045-11e9-9de6-904af6cec251.png)

____

Not sure what is going on, it looks like all paths normally don't have quotations.

**Comment by sampsyo** [comment]
OK! I am willing to bet this is a Windows-specific bug somewhere inside our `interactive_open` function in `util`. If anyone has both a Windows machine and some interest in digging deeper, it might be useful to start there and see how the call is being constructed.

**Comment by enzelbeltran** [comment]
I also did a bit more digging into the mpv repo and found a relevant issue (I think) from 5 years ago: [mpv.com not escaping spaces in executable path correctly #462](https://github.com/mpv-player/mpv/issues/462) and it was fixed in commit [```c849f1a```](https://github.com/mpv-player/mpv/commit/c849f1aa2353c8877b6dce58899f2dfd053fe808).

Also, I found the specific documentation in the [mpv manual](https://mpv.io/manual/master/#general-input-command-syntax) that relates to this issue:

> Arguments are separated by whitespace. This applies even to string arguments. For this reason, string arguments should be quoted with ". Inside quotes, C-style escaping can be used.

Thanks for your help!
